src/Evaluator.hs

```diff
@@ -135,6 +160,15 @@ data Kont where
   InDataCaseScrut :: ![(ConstructorPattern, Core)] -> !SrcLoc -> !VEnv -> !Kont -> Kont
   InTypeCaseScrut :: ![(TypePattern, Core)] -> !SrcLoc -> !VEnv -> !Kont -> Kont
 
+  {- Note [InCasePattern]
+     In case pattern is strictly not necessary, we could do this evaluation in
+     the host's runtime instead of in the evaluator but doing so would mean that
```

Comment:
what does "the host's runtime" mean?

Reply:
In cross-compilation the [host platform](https://wiki.osdev.org/Target_Triplet) is the system that the generated code will be run on. So all I'm saying here is that we just need to tag the continuation to know if the evaluator is in a primitive, in a case pattern or in a data case pattern. That tag does not strictly need to be in the evaluator, it could be part of the runtime, for example as a tag in some bits in the pointer to the data being evaluated.